lib/crawler.js

```diff
@@ -1062,98 +1062,97 @@ Crawler.prototype.queueURL = function(url, referrer) {
 
 */
 Crawler.prototype.fetchQueueItem = function(queueItem) {
-    var crawler = this;
-
-    // Variable declarations
-    var clientRequest,
-        timeCommenced;
-
-    // Mark as spooled
-    queueItem.status = "spooled";
-    var client = queueItem.protocol === "https" ? https : http;
-    var agent  = queueItem.protocol === "https" ? crawler.httpsAgent : crawler.httpAgent;
+    var crawler = this,
+        timeCommenced = Date.now();
 
-    // Up the socket limit if required.
-    if (agent.maxSockets < crawler.maxConcurrency) {
-        agent.maxSockets = crawler.maxConcurrency;
-    }
+    crawler.queue.update(queueItem.url, {
+        status: "spooled"
+    }, function(error, queueItem) {
```

Comment:
Ah, you're right - there are a few errors we need to handle. Just to clarify a few things about these errors:
I thought initially about retry strategies for the update method and if that was something we ought to have, but I realized that that's only needed for third party queue implementations, and they can handle retry strategies internally. 
So when we do get an error here, do we just throw it or do we emit some kind of event? We can't really go ahead with what we were doing in any case, and the risk of frequent errors from the update method should be very low (I don't even think the default method can return one), so maybe just throw it?